### ISSUE TITLE
Looks like thrown errors get reported inconsistently and incorrectly

### ISSUE THREAD

**Issue by lgandecki** [wallabyjs/public#2414]
### Issue description or question

Hello guys.

We've initially stumbled upon this with jest running through ts-node, but then I've made a simple repro with pure javascript and basically no setup besides installing jest [https://github.com/lgandecki/wallaby-awaited-errors-js/](https://github.com/lgandecki/wallaby-awaited-errors-js/), it behaved exactly the same way.

When I have a simple function that throws an error, the error shows up properly next to where it was thrown, and the function call inside test, but the dots don't turn red - acceptable, but not great UX (I often scan the dots to see where the error happens, and the gray color suggests the execution didn't get to that point.

![image](https://user-images.githubusercontent.com/4002543/82058278-d173b900-96c4-11ea-9da7-016fe29e3410.png)

When I have an async/await test with a timeout, it doesn't report an error at all.
It might be somehow connected with https://github.com/wallabyjs/public/issues/1746
I think this is a super crucial issue, as this is not really a rare edge case. This frequently gets me out of flow. This doesn't happen only with setTimeout, that's just a simple example. 

![image](https://user-images.githubusercontent.com/4002543/82058321-dd5f7b00-96c4-11ea-946b-ef4f5154b3ba.png)

We are preparing an example GraphQL setup with a lot of focus on tests and code generation, I can work around the above problem by messing with the stack trace in our testing tool, but then the dot still doesn't turn red. We want to convince a company that ordered that repo from us to start using wallabyjs, those awaited calls will be the main part of their work-flow and what we will show them. It would be fantastic to have this fixed in the next few weeks or at least have a deadline that we could point them to.
![image](https://user-images.githubusercontent.com/4002543/82057977-62966000-96c4-11ea-98db-4bbef88b4a81.png)

I feel like this worked better before.. 

```
{
  editorVersion: 'WebStorm 2020.1.1',
  pluginVersion: '1.0.173',
  editorType: 'IntelliJ',
  osVersion: 'darwin 19.4.0',
  nodeVersion: 'v12.16.2',
  coreVersion: '1.0.893',
  config: {
    diagnostics: {},
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    filesWithCoverageCalculated: [],
    globalSetup: false,
    micromatch: true,
    files: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true },
      {
        pattern: 'dist/|build/|coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /dist\/|build\/|coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true
      },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: true, trigger: true, load: true },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: 'dist/|build/|coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /dist\/|build\/|coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 2 },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 3 }
    ],
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    extensions: {},
    env: {
      type: 'node',
      params: {},
      runner: '<homeDir>/.nvm/versions/node/v12.16.2/bin/node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'auto.detect#1026773938'
  },
  packageJSON: { dependencies: {}, devDependencies: { jest: '^26.0.1' } },
  fs: { numberOfFiles: 3 },
  debug: [
    '2020-05-15T14:08:45.052Z config Detected Jest.\n',
    '2020-05-15T14:08:45.052Z config Configured Jest.\n',
    '2020-05-15T14:08:45.053Z project Wallaby Node version: v12.16.2\n',
    '2020-05-15T14:08:45.053Z project Wallaby config: <rootDir>/auto.detect\n',
    '2020-05-15T14:08:45.703Z project File cache: <homeDir>/Library/Caches/JetBrains/WebStorm2020.1/wallaby/projects/e29f50eedf9176f5\n',
    '2020-05-15T14:08:45.709Z uiService Listening port 51236\n',
    '2020-05-15T14:08:45.711Z project package.json file change detected, invalidating local cache\n',
    '2020-05-15T14:08:45.715Z workers Parallelism for initial run: 30, for regular run: 15\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #0\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #1\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #2\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #3\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #4\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #5\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #6\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #7\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #8\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #9\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #10\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #11\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #12\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #13\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #14\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #15\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #16\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #17\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #18\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #19\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #20\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #21\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #22\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #23\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #24\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #25\n',
    '2020-05-15T14:08:45.715Z workers Starting run worker instance #26\n',
    '2020-05-15T14:08:45.716Z workers Starting run worker instance #27\n',
    '2020-05-15T14:08:45.716Z workers Starting run worker instance #28\n',
    '2020-05-15T14:08:45.716Z workers Starting run worker instance #29\n',
    '2020-05-15T14:08:45.716Z workers Web server is listening at 52751\n',
    '2020-05-15T14:08:45.716Z project File cache requires some updates, waiting required files from IDE\n',
    '2020-05-15T14:08:45.830Z project Stopping process pool\n',
    '2020-05-15T14:08:45.832Z project Test run started; run priority: 3\n',
    '2020-05-15T14:08:45.833Z project Running all tests\n',
    '2020-05-15T14:08:45.835Z workers Starting test run, priority: 3\n',
    '2020-05-15T14:08:45.835Z nodeRunner Starting sandbox [worker #0, session #6hjbp]\n',
    '2020-05-15T14:08:45.835Z nodeRunner Preparing sandbox [worker #0, session #6hjbp]\n',
    '2020-05-15T14:08:45.860Z workers Started run worker instance (delayed) #1\n',
    '2020-05-15T14:08:45.861Z workers Started run worker instance (delayed) #0\n',
    '2020-05-15T14:08:45.861Z nodeRunner Prepared sandbox [worker #0, session #6hjbp]\n',
    '2020-05-15T14:08:45.861Z workers [worker #0, session #6hjbp] Running tests in sandbox\n',
    '2020-05-15T14:08:45.862Z workers Started run worker instance (delayed) #2\n',
    '2020-05-15T14:08:45.863Z workers Started run worker instance (delayed) #3\n',
    '2020-05-15T14:08:45.872Z workers Started run worker instance (delayed) #4\n',
    '2020-05-15T14:08:45.873Z workers Started run worker instance (delayed) #5\n',
    '2020-05-15T14:08:45.876Z workers Started run worker instance (delayed) #6\n',
    '2020-05-15T14:08:45.885Z workers Started run worker instance (delayed) #7\n',
    '2020-05-15T14:08:45.889Z workers Started run worker instance (delayed) #8\n',
    '2020-05-15T14:08:45.893Z workers Started run worker instance (delayed) #9\n',
    '2020-05-15T14:08:45.894Z workers Started run worker instance (delayed) #11\n',
    '2020-05-15T14:08:45.900Z workers Started run worker instance (delayed) #10\n',
    '2020-05-15T14:08:45.902Z workers Started run worker instance (delayed) #12\n',
    '2020-05-15T14:08:45.905Z workers Started run worker instance (delayed) #14\n',
    '2020-05-15T14:08:45.906Z workers Started run worker instance (delayed) #13\n',
    '2020-05-15T14:08:45.912Z workers Started run worker instance (delayed) #15\n',
    '2020-05-15T14:08:45.914Z workers Started run worker instance (delayed) #16\n',
    '2020-05-15T14:08:45.916Z workers Started run worker instance (delayed) #17\n',
    '2020-05-15T14:08:45.917Z workers Started run worker instance (delayed) #18\n',
    '2020-05-15T14:08:45.918Z workers Started run worker instance (delayed) #19\n',
    '2020-05-15T14:08:45.925Z workers Started run worker instance (delayed) #20\n',
    '2020-05-15T14:08:45.927Z workers Started run worker instance (delayed) #21\n',
    '2020-05-15T14:08:45.929Z workers Started run worker instance (delayed) #22\n',
    '2020-05-15T14:08:45.930Z workers Started run worker instance (delayed) #23\n',
    '2020-05-15T14:08:45.932Z workers Started run worker instance (delayed) #24\n',
    '2020-05-15T14:08:45.934Z workers Started run worker instance (delayed) #26\n',
    '2020-05-15T14:08:45.936Z workers Started run worker instance (delayed) #25\n',
    '2020-05-15T14:08:45.940Z workers Started run worker instance (delayed) #28\n',
    '2020-05-15T14:08:45.940Z workers Started run worker instance (delayed) #27\n',
    '2020-05-15T14:08:45.943Z workers Started run worker instance (delayed) #29\n',
    '2020-05-15T14:08:47.132Z workers [6hjbp] Loaded unknown number of test(s)\n',
    '2020-05-15T14:08:47.243Z workers [6hjbp] Test executed: async await error with changed stack\n',
    '2020-05-15T14:08:47.247Z workers [6hjbp] Test executed: normal error\n',
    '2020-05-15T14:08:47.248Z workers [6hjbp] Test executed: async await error\n',
    '2020-05-15T14:08:47.249Z workers [6hjbp] Run 1 test(s), skipped 0 test(s)\n',
    '2020-05-15T14:08:47.250Z workers [6hjbp] Sandbox is responsive, closing it\n',
    '2020-05-15T14:08:47.251Z project Test run finished\n',
    '2020-05-15T14:08:47.251Z project Processed console.log entries\n',
    '2020-05-15T14:08:47.251Z project Processed loading sequences\n',
    '2020-05-15T14:08:47.252Z project Processed executed tests\n',
    '2020-05-15T14:08:47.252Z project Processed code coverage\n',
    '2020-05-15T14:08:47.260Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by ArtemGovorov** [comment]
I think there may be an issue with your theme (the error source red colour seems to be replaced with some pale green colour, that is different from line-covered green - see line 32 vs line 36 on your second screenshot). Here is the result I'm getting with the default WS Darkcula:

<img width="845" alt="Screen Shot 2020-05-16 at 12 45 12 am" src="https://user-images.githubusercontent.com/979966/82063498-e6753a00-970e-11ea-86c7-00742bab4be7.png">

Changing your `Settings - Editor - Color Scheme - Wallaby.js - Error source` background color should fix it:

<img width="966" alt="Screen Shot 2020-05-16 at 12 49 12 am" src="https://user-images.githubusercontent.com/979966/82063809-4d92ee80-970f-11ea-8dcd-2c793a867ee5.png">

**Comment by lgandecki** [comment]
oh wow! What the heck! :) This must have changed today somehow when I was playing around with the reproduction - hence the "I feel like this worked better before.." comment. Very weird. I use wallabyjs every day and the erorrs have surely been red... 
Thanks so much for the instant help!

Anyway - no chance to have the problem nr 2 fixed/tweaked? I think that's #1746 .
It would be fantastic not having to hack the error stacks..

**Comment by ArtemGovorov** [comment]
Sorry, I have forgotten to address the second raised issue.

It doesn't look like Wallaby specific. There's very little magic in how Wallaby reports/maps errors - we need a stack trace of an error, and if some lines are in that stack, then we highlight those lines.

If you run your sample project in Jest CLI, you'll get the same result:

<img width="819" alt="Screen Shot 2020-05-16 at 10 45 05 am" src="https://user-images.githubusercontent.com/979966/82106351-b1df9d80-9763-11ea-8a6a-403edafd95de.png">

See how for the normal error there is a line of the test in the error stack to highlight. However, in the `async await error` test, there's no test line in the stack (apart from the line where the error is created).

Also, even a simpler example without Jest or Wallaby:

## a.js
```js
async function simpleAsyncThrowingFunction() {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            reject(new Error("async await error"));
        });
    });
}

(async () => {
    try {
        await simpleAsyncThrowingFunction();
    }
    catch (e) {
        console.log(e)
    }
})();
```

running

```
node a.js
```

produces:

```
Error: async await error
    at Timeout._onTimeout (./a.js:4:20)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
```

The line with `await simpleAsyncThrowingFunction();` is simply not in the reported error stack.

One solution would be to create the error outside of the timeout (generally where its stack will have the original test line), for example:


```js
async function simpleAsyncThrowingFunction() {
    var err = new Error("async await error")
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            reject(err);
        });
    });
}

(async () => {
    try {
        await simpleAsyncThrowingFunction();
    }
    catch (e) {
        console.log(e)
    }
})();
```

produces:

```
Error: async await error
    at simpleAsyncThrowingFunction (./a.js:2:15)
    at ./a.js:12:15 <----
    at Object.<anonymous> (./a.js:17:3)
    at Module._compile (internal/modules/cjs/loader.js:1147:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
    at Module.load (internal/modules/cjs/loader.js:996:32)
    at Function.Module._load (internal/modules/cjs/loader.js:896:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
```
